test/boost/sstable_compaction_test.cc

```diff
@@ -210,18 +210,10 @@ SEASTAR_TEST_CASE(compaction_manager_basic_test) {
     auto s = make_shared_schema({}, some_keyspace, some_column_family,
         {{"p1", utf8_type}}, {{"c1", utf8_type}}, {{"r1", int32_type}}, {}, utf8_type);
 
-    auto cm = compaction_manager_for_testing();
-
     auto tmp = tmpdir();
-    replica::column_family::config cfg = env.make_table_config();
-    cfg.datadir = tmp.path().string();
-    cfg.enable_commitlog = false;
-    cfg.enable_incremental_backups = false;
-    auto cl_stats = make_lw_shared<cell_locker_stats>();
-    auto tracker = make_lw_shared<cache_tracker>();
-    auto cf = make_lw_shared<replica::column_family>(s, cfg, replica::column_family::no_commitlog(), *cm, env.manager(), *cl_stats, *tracker);
-    cf->start();
-    cf->mark_ready_for_writes();
+    column_family_for_tests cf(env.manager(), s, tmp.path().string());
+    auto& cm = cf.get_compaction_manager();
+    auto close_cf = deferred_stop(cf);
```

Comment:
With table_state, we can decouple compaction tests from replica::table completely.

Reply:
That's indeed the goal.